core/src/main/java/hudson/ClassicPluginStrategy.java

```diff
@@ -300,6 +330,47 @@ private DetachedPlugin(String shortName, String splitWhen, String requireVersion
             this.requireVersion = requireVersion;
         }
 
+        /**
+         * Get the short name of the plugin.
+         * @return The short name of the plugin.
+         * @since FIXME
+         */
+        public String getShortName() {
+            return shortName;
+        }
+
+        /**
+         * Get the Jenkins version from which the plugin was detached.
+         * @return The Jenkins version from which the plugin was detached.
+         * @since FIXME
+         */
+        public VersionNumber getSplitWhen() {
+            return splitWhen;
+        }
+
+        /**
+         * Get the minimum required version of the plugin.
+         * @return The minimum required version of the plugin.
+         * @since FIXME
+         */
+        public String getRequireVersion() {
+            return requireVersion;
+        }
+
+        /**
+         * Convert a collection of DetachedPlugins to a Set of plugins names.
+         * @param detachedPlugins The collection of DetachedPlugins.
+         * @return The list of plugin names.
+         * @since FIXME
+         */
+        public static Set<String> toPluginNameList(@Nonnull Collection<ClassicPluginStrategy.DetachedPlugin> detachedPlugins) {
+            Set<String> pluginNames = new LinkedHashSet<>();
+            for (ClassicPluginStrategy.DetachedPlugin detachedPlugin : detachedPlugins) {
+                pluginNames.add(detachedPlugin.getShortName());
+            }
+            return pluginNames;
```

Comment:
Because this method is just adding conceptual weight without really adding anything you could not do trivially without it. It is not even called from any new code in this PR that I can see! And APIs without demonstrated uses are always a bad idea.